spyderlib/widgets/arrayeditor.py

```diff
@@ -428,6 +437,33 @@ def copy(self):
         cliptxt = self._sel_to_text( self.selectedIndexes() )
         clipboard = QApplication.clipboard()
         clipboard.setText(cliptxt)
+       
+    def load_all_rows(self):
+        while self.model().can_fetch_more(rows=True):
+            self.model().fetch_more(rows=True)
+            
+    @Slot(int)
+    def load_all_rows_select_column(self, column):
+        self.load_all_rows()
+        self.selectColumn(column)
+
+    def load_all_columns(self):
+        while self.model().can_fetch_more(columns=True):
+            self.model().fetch_more(columns=True)
+            
+    @Slot(int)
+    def load_all_columns_select_row(self, row):
+        self.load_all_columns()
+        self.selectRow(row)
+    
+    def selectAll(self):
+        """If we select everything then we probably intended to 
+        select the entire array (and hence must load any unloaded data)"""

```

Comment:
Here you need to add the warning I mentioned (using a `QMessageBox`). It should say something like:

> You are trying to select the whole contents of this array, which has %d elements. If you continue with this operation, it could make Spyder crash or consume a large amount of RAM.
> Do you want to do it?

Where `%d` is going to be replaced by the total number of elements of the array.